docker/Dockerfile.multiarch

```diff
@@ -0,0 +1,55 @@
+# FIXME: use "--platform=$BUILDPLATFORM" and solve openssl cross-compile issue
+FROM rust:1.67.0-alpine as builder
+
+# Install build dependencies
+RUN apk add --no-cache git openssl-dev libpq-dev musl-dev
+
+# Set the working directory to /app and copy the source code
+WORKDIR /app
+COPY . .
+
+# Set the target architecture (can be set using --build-arg), buildx set it automatically
+ARG TARGETARCH
+
+# This can be set to release using --build-arg
+ARG RUST_RELEASE_MODE="debug"
+
+# Prepare toolchain
+# Docker and Rust use different architecture naming schemas, so we have to convert them
+RUN case $TARGETARCH in \
+      arm64) RUSTARCH=aarch64 ;; \
+      amd64) RUSTARCH=x86_64 ;; \
+      *) echo "unsupported architecture: $TARGETARCH"; exit 3 ;; \
+    esac \
+ && echo "RUSTARCH=$RUSTARCH" >> .buildenv
+
+# Debug mode build
+RUN --mount=type=cache,target=/app/target \
+    if [ "$RUST_RELEASE_MODE" = "debug" ]; then \
+      source .buildenv \
+      && echo "pub const VERSION: &str = \"$(git describe --tag)\";" > "crates/utils/src/version.rs" \
+      && rustup target add ${RUSTARCH}-unknown-linux-musl \
+      && cargo build --target ${RUSTARCH}-unknown-linux-musl \
+      && cp ./target/${RUSTARCH}-unknown-linux-musl/${RUST_RELEASE_MODE}/lemmy_server /app/lemmy_server; \
+    fi
+
+# Release mode build
+RUN \
+    if [ "$RUST_RELEASE_MODE" = "release" ]; then \
+      source .buildenv \
+      && rustup target add ${RUSTARCH}-unknown-linux-musl \
+      && cargo build --target ${RUSTARCH}-unknown-linux-musl --release \
+      && cp ./target/${RUSTARCH}-unknown-linux-musl/${RUST_RELEASE_MODE}/lemmy_server /app/lemmy_server; \
+    fi
+
+# The Alpine runner
+FROM alpine:3 as lemmy
+
+# Install libpq for Postgres
+RUN apk add --no-cache ca-certificates libpq
+
+# Copy resources
+COPY --from=builder /app/lemmy_server /app/lemmy
+
+EXPOSE 8536
+CMD ["/app/lemmy"]
```

Comment:
Should probably remove this file as cross-compile isnt working. And based on the age of the issue it wont get fixed anytime soon :(

Reply:
This pull does not address cross compilation!

Its moved in a extra pull

Reply:
Looks like this was added to the main branch anyway.

Reply:
the multiarch docker image is just ment to be build with buildx ... and the cross-compiline is just an optional feature ... for now I would recomend to just use it as single source of truth and have different agents that target this dockerfile

the old Dockerfile is just for normal devs who wana build the image but dont have [buildx](https://docs.docker.com/engine/reference/commandline/buildx/) installed

Reply:
In that case I would prefer to remove the old dockerfile, and list buildx as a requirements. cc @dessalines

Reply:
Yep, I'm def fine with removing the old `Dockerfile`

Reply:
Well the "old" is the same as the new one ... just without some fancy if statements and args on FROM 😆. so you can make the multiarch to a normal one again at any time

Reply:
#2818

Reply:
-> https://github.com/LemmyNet/lemmy/pull/2818#issuecomment-1519256243